src/foam/lang/types.js

```diff
@@ -1085,7 +1085,7 @@ foam.CLASS({
     {
       name: 'adapt',
       value: function(oldValue, newValue, prop) {
-        return prop.of.isInstance(newValue) ?
+        return prop?.of.isInstance(newValue) ?
```

Comment:
Is this correct? 
Should it be:
return (prop || foam.lang.FObject).of...

Reply:
(prop || this) works to ensure we always call isInstance